packages/cli-platform-ios/src/tools/pods.ts

```diff
@@ -0,0 +1,80 @@
+import path from 'path';
+import fs from 'fs-extra';
+import {createHash} from 'crypto';
+import chalk from 'chalk';
+import {
+  CLIError,
+  cacheManager,
+  getLoader,
+} from '@react-native-community/cli-tools';
+import installPods from './installPods';
+
+interface ResolvePodsOptions {
+  forceInstall?: boolean;
+}
+
+export function getPackageJson(root: string) {
+  return require(path.join(root, 'package.json'));
+}
+
+export function normalizeDependencies(dependencies: Record<string, string>) {
+  return Object.entries(dependencies)
+    .map(([name, version]) => `${name}@${version}`)
+    .sort();
+}
+
+export function dependenciesToString(dependencies: string[]) {
+  return dependencies.join('\n');
+}
+
+export function generateMd5Hash(text: string) {
+  return createHash('md5').update(text).digest('hex');
+}
+
+export function compareMd5Hashes(hash1: string, hash2: string) {
+  return hash1 === hash2;
+}
+
+export default async function resolvePods(
+  root: string,
+  options?: ResolvePodsOptions,
+) {
+  const packageJson = getPackageJson(root);
+  const podsPath = path.join(root, 'ios', 'Pods');
+  const arePodsInstalled = fs.existsSync(podsPath);
+  const dependencies = normalizeDependencies({
+    ...packageJson.dependencies,
+    ...packageJson.devDependencies,
+  });
```

Comment:
instead of looping through all dependencies and make any tiny non-ios related dep invalidate our pods cache, we could rather use `config.depdenencies` and filter out those that have `platforms.ios` set